### PR TITLE
Install catkin-pkg

### PR DESCRIPTION
**Pull request by eeberhard**
* Explicitly install the catkin packages for the python3 interpreter.

When installing python3 from source as the default system interpreter, the catkin-pkg module could not be found by ament_cmake, which is invoked during colcon build. This naturally caused all downstream ROS2 workspace builds to fail.

Now colcon build succeeds (tested locally to build modulo image with most recent ghcr.io/aica-technology/ros2-control-libraries:galactic)

### CODE REVIEW

**Review by domire8 — APPROVED** [review]
Thanks for the sudo fix, that was what I was going to propose. After #30, I'm happy if this goes in, I can confirm all the images build now

**Review by domire8 — CHANGES_REQUESTED** [review]
Actually, cancel that, The ros2_modulo_control image doesn't build

**Comment by eeberhard** [comment]
very well, and so we find our way down to the next issue. Indeed I only tested the modulo stage. What is the error output you got from the subsequent ros2 control build attempt?

File: ros2_ws/Dockerfile [hunk 1]
```diff
@@ -52,11 +52,13 @@ ARG PYTHON_VER=3.10.2
...
+RUN tar -xf Python-${PYTHON_VER}.tar.xz
 WORKDIR Python-${PYTHON_VER}
-RUN sudo ./configure --enable-optimizations --with-ensurepip=install \
-  && sudo make -j8 \
-  && sudo make install
+RUN ./configure --enable-optimizations --with-ensurepip=install \
+  && make -j8 \
+  && make install
+
+RUN python3 -m pip install catkin_pkg
```
Comment (domire8) [inline ros2_ws/Dockerfile:61]:
```suggestion
RUN ./configure --enable-optimizations --with-ensurepip=install --enable-shared \
  && make -j8 \
  && make install
RUN ldconfig

RUN python3 -m pip install catkin_pkg empy numpy lark
```

**Review by domire8 — COMMENTED** [review]
With those changes below, even ros2 modulo control builds now, you just get warnings like this for several packages:
```
#15 141.3 --- stderr: controller_manager
#15 141.3 <string>:1: DeprecationWarning: The distutils package is deprecated and slated for removal in Python 3.12. Use setuptools or check PEP 632 for potential alternatives
#15 141.3 <string>:1: DeprecationWarning: The distutils.sysconfig module is deprecated, use sysconfig instead
#15 141.3 ---
``

**Comment by eeberhard** [comment]
ROS2 Galactic expects Python 3.8, and it turns out that it is harder to upgrade than I expected. With 3.9.X, the backend builds (all components and state engine). However, it still fails at runtime. Specifically, it cannot resolve the `rosidl_typesupport_c` runtime library.

With what I've tried so far, `colcon build` will still make a mess and install some things to `/path/to/workspace/install/package_name/lib/python3.8/site-packages` and `/path/to/workspace/install/other_package_name/lib/python3.9/site-packages` on the PYTHONPATH. 

This is probably fixable but needs to be put on hold until it is more stable. Until then we must revert the original #28 or risk our downstream targets continuing to fail.

**Pull request closed without merging**